pkgsStatic.buildPackages.haskell.compiler.ghc96 fail to build

```
@nix { "action": "setPhase", "phase": "buildPhase" }
building
hadrianFlags: --flavour=release+fully_static+no_dynamic_ghc+split_sections --bignum=gmp --docs=no-sphinx -j4
]0;Starting...[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "binary", pkgPath = "libraries/binary"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "bytestring", pkgPath = "libraries/bytestring"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "Cabal-syntax", pkgPath = "libraries/Cabal/Cabal-syntax"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "Cabal", pkgPath = "libraries/Cabal/Cabal"})))[0m
[34m[0m| Successfully generated compiler/ghc.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghc", pkgPath = "compiler"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "containers", pkgPath = "libraries/containers/containers"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "directory", pkgPath = "libraries/directory"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "process", pkgPath = "libraries/process"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "exceptions", pkgPath = "libraries/exceptions"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "filepath", pkgPath = "libraries/filepath"})))[0m
[34m[0m| Successfully generated libraries/ghc-boot/ghc-boot.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghc-boot", pkgPath = "libraries/ghc-boot"})))[0m
[34m[0m| Successfully generated libraries/ghc-boot-th/ghc-boot-th.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghc-boot-th", pkgPath = "libraries/ghc-boot-th"})))[0m
[34m[0m| Successfully generated libraries/ghc-heap/ghc-heap.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghc-heap", pkgPath = "libraries/ghc-heap"})))[0m
[34m[0m| Successfully generated libraries/ghci/ghci.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghci", pkgPath = "libraries/ghci"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "hpc", pkgPath = "libraries/hpc"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "mtl", pkgPath = "libraries/mtl"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "parsec", pkgPath = "libraries/parsec"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "time", pkgPath = "libraries/time"})))[0m
[34m[0m| Successfully generated libraries/template-haskell/template-haskell.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "template-haskell", pkgPath = "libraries/template-haskell"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "text", pkgPath = "libraries/text"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "transformers", pkgPath = "libraries/transformers"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "unix", pkgPath = "libraries/unix"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "array", pkgPath = "libraries/array"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "base", pkgPath = "libraries/base"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "deepseq", pkgPath = "libraries/deepseq"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghc-bignum", pkgPath = "libraries/ghc-bignum"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghc-compact", pkgPath = "libraries/ghc-compact"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "ghc-prim", pkgPath = "libraries/ghc-prim"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "haskeline", pkgPath = "libraries/haskeline"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "integer-gmp", pkgPath = "libraries/integer-gmp"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "pretty", pkgPath = "libraries/pretty"})))[0m
[34m[0m| Successfully generated rts/rts.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "rts", pkgPath = "rts"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "stm", pkgPath = "libraries/stm"})))[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Library, pkgName = "xhtml", pkgPath = "libraries/xhtml"})))[0m
[34m[0m| Configure package 'bytestring'[0m[0m
[34m[0m| Configure package 'ghc-prim'[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Program, pkgName = "hp2ps", pkgPath = "utils/hp2ps"})))[0m
[34m[0m| Configure package 'ghc-heap'[0m[0m
[34m[0m| Configure package 'parsec'[0m[0m
[34m[0m| Configure package 'hpc'[0m[0m
[34m[0m| Configure package 'deepseq'[0m[0m
[34m[0m| Configure package 'ghc-boot-th'[0m[0m
[34m[0m| Configure package 'rts'[0m[0m
[34m[0m| Configure package 'process'[0m[0m
[34m[0m| Configure package 'integer-gmp'[0m[0m
[34m[0m| Configure package 'array'[0m[0m
[34m[0m| Configure package 'time'[0m[0m
[34m[0m| Configure package 'unix'[0m[0m
[34m[0m| Configure package 'mtl'[0m[0m
[34m[0m| Configure package 'transformers'[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Program, pkgName = "unlit", pkgPath = "utils/unlit"})))[0m
[34m[0m| Configure package 'stm'[0m[0m
[34m[0m| Configure package 'ghci'[0m[0m
[34m[0m| Configure package 'directory'[0m[0m
[34m[0m| Copy file: driver/ghc-usage.txt => _build/stage1/lib/ghc-usage.txt[0m[0m
[34m[0m| Configure package 'containers'[0m[0m
[34m[0m| Configure package 'binary'[0m[0m
[34m[0m| Configure package 'Cabal'[0m[0m
[34m[0m| Configure package 'ghc-compact'[0m[0m
[34m[0m| Configure package 'text'[0m[0m
[34m[0m| Configure package 'Cabal-syntax'[0m[0m
[34m[0m| Configure package 'pretty'[0m[0m
[34m[0m| Configure package 'hp2ps'[0m[0m
[34m[0m| Configure package 'xhtml'[0m[0m
[34m[0m| System GMP library/framework will be used[0m[0m
[34m[0m| Configure package 'filepath'[0m[0m
[34m[0m| Configure package 'base'[0m[0m
[34m[0m| Copy file: driver/ghci-usage.txt => _build/stage1/lib/ghci-usage.txt[0m[0m
[34m[0m| Successfully generated ghc/ghc-bin.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Program, pkgName = "ghc-bin", pkgPath = "ghc"})))[0m
[34m[0m| Configure package 'exceptions'[0m[0m
[34m[0m| Configure package 'unlit'[0m[0m
[34m[0m| Successfully generated utils/ghc-pkg/ghc-pkg.cabal from its template[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Program, pkgName = "ghc-pkg", pkgPath = "utils/ghc-pkg"})))[0m
[34m[0m| Configure package 'haskeline'[0m[0m
[34m[0m| Run GhcPkg Recache (Stage0 InTreeLibs): none => none[0m[0m
[34m[0m| Copy file: mk/system-cxx-std-lib-1.0.conf => _build/stage1/lib/package.conf.d/system-cxx-std-lib-1.0.conf[0m[0m
[34m[0m| Copy file: llvm-passes => _build/stage1/lib/llvm-passes[0m[0m
[34m[0m| Copy file: llvm-targets => _build/stage1/lib/llvm-targets[0m[0m
[34m[0m| Configure package 'template-haskell'[0m[0m
[34m# cabal-read (for OracleQ (PackageDataKey (Package {pkgType = Program, pkgName = "hsc2hs", pkgPath = "utils/hsc2hs"})))[0m
[34m[0m| Copy file: libraries/ghc-bignum/gmp/ghc-gmp.h => _build/stage1/libraries/ghc-bignum/build/include/ghc-gmp.h[0m[0m
[34m[0m| Configure package 'ghc-bignum'[0m[0m
[34m[0m| Configure package 'ghc-boot'[0m[0m
[34m[0m| Copy file: utils/hsc2hs/data/template-hsc.h => _build/stage1/lib/template-hsc.h[0m[0m
[34m[0m| Successfully generated _build/stage1/lib/settings.[0m[0m
[34m# cabal-configure (for _build/stage0/utils/unlit/setup-config)[0m
[34m[0m| Configure package 'ghc'[0m[0m
[34m[0m| Configure package 'ghc-boot'[0m[0m
[34m[0m| Configure package 'exceptions'[0m[0m
[34m[0m| Configure package 'bytestring'[0m[0m
[34m[0m| Configure package 'mtl'[0m[0m
[34m[0m| Configure package 'unlit'[0m[0m
[34m[0m| Run GhcPkg Recache (Stage0 GlobalLibs): none => none[0m[0m
[34m[0m| Configure package 'transformers'[0m[0m
[34m[0m| Configure package 'unix'[0m[0m
[34m[0m| Configure package 'ghc-boot-th'[0m[0m
[34m[0m| Configure package 'time'[0m[0m
[34m# cabal-configure (for _build/stage0/libraries/ghc-boot-th/setup-config)[0m
[34m[0m| Copy file: driver/ghc-usage.txt => _build/stage0/lib/ghc-usage.txt[0m[0m
[34m[0m| Copy file: driver/ghci-usage.txt => _build/stage0/lib/ghci-usage.txt[0m[0m
[34m[0m| Configure package 'Cabal-syntax'[0m[0m
[34m[0m| Configure package 'ghc-heap'[0m[0m
[34m# cabal-configure (for _build/stage0/libraries/time/setup-config)[0m
[34m# cabal-configure (for _build/stageBoot/utils/unlit/setup-config)[0m
[34m[0m| Configure package 'containers'[0m[0m
[34m[0m| Configure package 'template-haskell'[0m[0m
[34m# cabal-autogen (for _build/stage0/libraries/ghc-boot-th/build/autogen/cabal_macros.h)[0m
[34m[0m| Configure package 'binary'[0m[0m
[34m[0m| Run Cc (FindCDependencies CDep) (Stage0 InTreeLibs): utils/unlit/unlit.c => _build/stage0/utils/unlit/build/c/unlit.o.d[0m[0m
[34m[0m| Copy file: mk/system-cxx-std-lib-1.0.conf => _build/stage0/lib/package.conf.d/system-cxx-std-lib-1.0.conf[0m[0m
[34m[0m| Copy file: llvm-targets => _build/stage0/lib/llvm-targets[0m[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): base[0m[0m
[34m[0m| Copy file: llvm-passes => _build/stage0/lib/llvm-passes[0m[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): base[0m[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): ghc-prim[0m[0m
[34m# cabal-autogen (for _build/stage0/utils/unlit/build/unlit/autogen/cabal_macros.h)[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): ghc-bignum[0m[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): ghc-prim[0m[0m
[34m[0m| Run Cc (FindCDependencies CDep) (Stage0 InTreeLibs): utils/unlit/fs.c => _build/stage0/utils/unlit/build/c/fs.o.d[0m[0m
[34m[0m| Run Ghc CompileCWithGhc (Stage0 InTreeLibs): utils/unlit/unlit.c => _build/stage0/utils/unlit/build/c/unlit.o[0m[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): ghc-bignum[0m[0m
[34m[0m| Run Ghc CompileCWithGhc (Stage0 InTreeLibs): utils/unlit/fs.c => _build/stage0/utils/unlit/build/c/fs.o[0m[0m
[34m[0m| Configure package 'parsec'[0m[0m
[34m[0m| Successfully generated _build/stage0/lib/settings.[0m[0m
[34m[0m| Configure package 'ghc'[0m[0m
[34m[0m| Configure package 'hpc'[0m[0m
[34m[0m| Configure package 'ghci'[0m[0m
[34m[0m| Configure package 'Cabal'[0m[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): rts[0m[0m
[34m[0m| Run GhcPkg Unregister (Stage0 InTreeLibs): rts => none[0m[0m
[34m# cabal-configure (for _build/stage0/libraries/transformers/setup-config)[0m
[34m[0m| Run Cc (FindCDependencies CDep) (Stage0 GlobalLibs): utils/unlit/unlit.c => _build/stageBoot/utils/unlit/build/c/unlit.o.d[0m[0m
[34m# cabal-autogen (for _build/stageBoot/utils/unlit/build/unlit/autogen/cabal_macros.h)[0m
[34m[0m| Run Ghc CompileCWithGhc (Stage0 GlobalLibs): utils/unlit/unlit.c => _build/stageBoot/utils/unlit/build/c/unlit.o[0m[0m
[34m[0m| Run GhcPkg Recache (Stage0 InTreeLibs): none => none[0m[0m
[34m[0m| Run GhcPkg Dependencies (Stage0 InTreeLibs): rts[0m[0m
[34m[0m| Run GhcPkg Unregister (Stage0 InTreeLibs): rts => none[0m[0m
[34m[0m| Configure package 'process'[0m[0m
[34m[0m| Configure package 'directory'[0m[0m
[34m# cabal-configure (for _build/stage0/utils/hp2ps/setup-config)[0m
[34m[0m| Run GhcPkg Copy (Stage0 InTreeLibs): rts => _build/stage0/lib/package.conf.d/rts.conf[0m[0m
[34m[0m| Run Ghc LinkHs (Stage0 InTreeLibs): _build/stage0/utils/unlit/build/c/fs.o (and 1 more) => _build/stage0/bin/x86_64-unknown-linux-musl-unlit[0m[0m
[31mCommand line: /nix/store/4gqivrz4j75995gpf2qr930sgscp9yz6-ghc-binary-9.2.4/bin/ghc -Wall -Wcompat -hisuf hi -osuf o -hcsuf hc -static -hide-all-packages -no-user-package-db '-package-env -' '-package-db _build/stage0/lib/package.conf.d' '-this-unit-id unlit-0.1' -i -i/build/ghc-a5bd0eb/_build/stage0/utils/unlit/build -i/build/ghc-a5bd0eb/_build/stage0/utils/unlit/build/unlit/autogen -i/build/ghc-a5bd0eb/utils/unlit -I_build/stage0/utils/unlit/build -optP-include -optP_build/stage0/utils/unlit/build/unlit/autogen/cabal_macros.h -outputdir _build/stage0/utils/unlit/build -fdiagnostics-color=always -no-auto-link-packages -no-hs-main -optc-Wno-error=inline -optP-Wno-nonportable-include-path _build/stage0/utils/unlit/build/c/fs.o _build/stage0/utils/unlit/build/c/unlit.o -o _build/stage0/bin/x86_64-unknown-linux-musl-unlit -O -H64m -XHaskell2010 -optl -static[0m
[31m===> Command failed with error code: 1[0m
/nix/store/frh9l9nrdysasdi2gs7i241s241ngjw2-binutils-2.39/bin/ld: cannot find -lc: No such file or directory
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
]0;Finished in 2.34sCommand failed
[31mBuild failed.[0m
```

Full log: 
[static-ghc-head-log.txt](https://github.com/NixOS/nixpkgs/files/10340769/static-ghc-head-log.txt)

I wasn't going to say anything, but there is a small chance it's relevant. I have a devenv (not pure nix) project I was trying to build with ghc 9.8 today and despite having postgresql in the dependencies I get:

```
`cc' failed in phase `Linker'. (Exit code: 1) /nix/store/v71nlgkf8g6d36gac9r44pmlx54mdql3-binutils-2.41/bin/ld.gold: error: cannot find -lpq
```

It seems possible this is related to dynamic/static libs, but I haven't tried reproducing with vanilla nix to verify.

--- a/pkgs/development/compilers/ghc/common-hadrian.nix
+++ b/pkgs/development/compilers/ghc/common-hadrian.nix
@@ -13,6 +13,7 @@
 , pkgsBuildTarget
 , pkgsHostTarget
 , targetPackages
+, fetchpatch
 
 # build-tools
 , bootPkgs
@@ -169,6 +170,13 @@
           (if lib.versionAtLeast version "9.8"
            then ./docs-sphinx-7-ghc98.patch
            else ./docs-sphinx-7.patch )
+        ]
+        ++ lib.optionals (lib.versionAtLeast version "9.6" && lib.versionOlder version "9.8") [
+          (fetchpatch {
+            name = "fix-fully_static.patch";
+            url = "https://gitlab.haskell.org/ghc/ghc/-/commit/1bb24432ff77e11a0340a7d8586e151e15bba2a1.diff";
+            hash = "sha256-MpvTmFFsNiPDoOp9BhZyWeapeibQ77zgEV+xzZ1UAXs=";
+          })
         ]
         ++ lib.optionals (stdenv.targetPlatform.isDarwin && stdenv.targetPlatform.isAarch64) [
           # Prevent the paths module from emitting symbols that we don't use
@@ -555,10 +563,6 @@ stdenv.mkDerivation ({
     ] ++ lib.teams.haskell.members;
     timeout = 24 * 3600;
     inherit (ghc.meta) license platforms;
-    # https://github.com/NixOS/nixpkgs/issues/208959
-    broken =
-      (lib.versionAtLeast version "9.6" && lib.versionOlder version "9.8")
-      && stdenv.targetPlatform.isStatic;
   };
 
   dontStrip = targetPlatform.useAndroidPrebuilt || targetPlatform.isWasm;

